operations/kubernetes/Troubleshoot_Postgres_Database.md

```diff
@@ -175,8 +175,87 @@ postgres@keycloak-postgres-2:~$ patronictl list
++-------------------+---------------------+--------------+--------+---------+----+-----------+
+| keycloak-postgres | keycloak-postgres-0 | 10.42.10.22  |        | running | 47 |         0 |
+| keycloak-postgres | keycloak-postgres-1 | 10.40.11.191 | Leader | running | 47 |           |
+| keycloak-postgres | keycloak-postgres-2 | 10.40.11.190 |        | stopped |    |   unknown |
++-------------------+---------------------+--------------+--------+---------+----+-----------+
+```
+ 
+Exec into that pod that is `stopped` and check the most recent postgres log for any `invalid segment number 0` errors relating to pg_internal.init.* files.
+
+```bash
+ncn-w001# kubectl exec keycloak-postgres-2 -n services -it -- bash
+postgres@keycloak-postgres-2:~$ export LOG=`ls -t /home/postgres/pgdata/pgroot/pg_log/*.csv | head -1`
+```
+
+Example output:
+
+```
+postgres@keycloak-postgres-2:~$ grep pg_internal.init $LOG | grep "invalid segment number 0" | tail -1
+2022-02-01 16:59:35.529 UTC,"standby","",227600,"127.0.0.1:42264",61f966f7.37910,3,"sending backup ""pg_basebackup base backup""",2022-02-01 16:59:35 UTC,7/0,0,ERROR,XX000,"invalid segment number 0 in file ""pg_internal.init.2239188""",,,,,,,,,"pg_basebackup"
+```
+
+If the check above finds such files, first find and delete any zero length pg_internal.init.* files. (Double check the syntax before running the step to delete these files with `-size 0 -exec rm {} \;`.)
+
+Example output:
+
+```bash
+postgres@keycloak-postgres-2:~$ find /home/postgres/pgdata -name pg_internal.init.* -size 0 
+./pgroot/data/base/16622/pg_internal.init.2239004
+...
+./pgroot/data/base/16622/pg_internal.init.2239010
+
+postgres@keycloak-postgres-2:~$ find /home/postgres/pgdata -name pg_internal.init.* -size 0 -exec rm {} \;
+```
+
+Then find and delete (or move to a different location) any non-zero length pg_internal.init.* files that were truncated when the file system filled up.
+
+Example output:
+
+```bash
+postgres@keycloak-postgres-2:~$ grep pg_internal.init $LOG | grep "invalid segment number 0" | tail -1
+2022-02-01 16:59:35.529 UTC,"standby","",227600,"127.0.0.1:42264",61f966f7.37910,3,"sending backup ""pg_basebackup base backup""",2022-02-01 16:59:35 UTC,7/0,0,ERROR,XX000,"invalid segment number 0 in file ""pg_internal.init.2239188""",,,,,,,,,"pg_basebackup"
+
+postgres@keycloak-postgres-2:~$ find ~/pgdata -name pg_internal.init.2239188
+/home/postgres/pgdata/pgroot/data/base/16622/pg_internal.init.2239188
+
+postgres@keycloak-postgres-2:~$ rm /home/postgres/pgdata/pgroot/data/base/16622/pg_internal.init.2239188
+```
+
+Iterate over the above step until there are no more new `invalid segment number 0` messages. At this point, verify that the cluster member has started.
+
+```bash
+ncn-w001# kubectl exec keycloak-postgres-0 -c postgres -n services -it -- bash
+```
+
+Example output:
+
+```
+postgres@keycloak-postgres-2:~$ patronictl list
```

Comment:
the command just above used keycloak-postgres-0 so this would also have keycloak-postgres-0

Reply:
fixed the typo - tx